object/state.go

```diff
@@ -63,7 +124,7 @@ func (e *Environment) Set(name string, val Object) Object {
 		old, ok := e.Get(name) // not ok
 		if ok {
 			log.Infof("Attempt to change constant %s from %v to %v", name, old, val)
-			if !Hashable(old) || !Hashable(val) || old != val {
+			if Equals(old, val) == FALSE {
```

Comment:
The uppercase false caught my attention. Shouldn't be this?

```suggestion
			if !Equals(old, val) {
```

Reply:
Equals() is returning an Object and thus TRUE or FALSE not a boolean directly (there is conversion function)

(because it's used in the `==` operator for instance where objects are needed not go primitive types)

Reply:
maybe I can do the wrapping elsewhere though/will check

Reply:
but you're right, it was ugly, changed to do NativeBoolToBooleanObject() only in leaves/where needed

thanks!